[Kinjal Agarwal] iP

kinjalagarwal1810 opened the pull request:
### DukeApp

> "You can do anything you set your mind to" - Ben Franklin

- text-based
- easy to learn
- _SUPER FAST_ to use!

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2324S2/ip/pull/270)
2. double-click it
3. add your tasks
4. let it manage your tasks for you 😃

And its **FREE**!

_Features:_

- [X] Manage tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If your a Java Programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

kinjalagarwal1810 committed:
Level 0. Rename, Greet, Exit

kinjalagarwal1810 committed:
Level-1

kinjalagarwal1810 committed:
Level-2

kinjalagarwal1810 committed:
Level-2

kinjalagarwal1810 committed:
Level-3

kinjalagarwal1810 committed:
Level-4

kinjalagarwal1810 committed:
Level-5

kinjalagarwal1810 committed:
A-TextUiTesting

kinjalagarwal1810 committed:
Level-5

kinjalagarwal1810 committed:
Level-6

kinjalagarwal1810 committed:
Level-7 (incomplete)

kinjalagarwal1810 committed:
Level-7

kinjalagarwal1810 committed:
Merge branch 'branch-Level-7'

kinjalagarwal1810 committed:
Level-8

kinjalagarwal1810 committed:
Merge branch 'branch-Level-8'

kinjalagarwal1810 committed:
A-MoreOOP

kinjalagarwal1810 committed:
A-Packages

kinjalagarwal1810 committed:
Merge branch 'A-Packages'

kinjalagarwal1810 committed:
Merge branch 'add-gradle-support' of https://github.com/kinjalagarwal1810/ip

kinjalagarwal1810 committed:
A-JUnit

kinjalagarwal1810 committed:
A-Jar

kinjalagarwal1810 committed:
A-JavaDoc

kinjalagarwal1810 committed:
A-JavaDoc

kinjalagarwal1810 committed:
Implement A-CodingStandard

kinjalagarwal1810 committed:
Implement Level-9

kinjalagarwal1810 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/command/Command.java

kinjalagarwal1810 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Ui.java

kinjalagarwal1810 committed:
Fix errors from merge

0-yibai commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,112 @@
+                default: {
+                    throw new JamieException("Error occurred when reading data from storage file.\n"
+                            + "Therefore, creating a new task list.");
+                }
```
Perhaps remove the indent before 'case'?
According to our coding standard:
![Screenshot 2024-02-07 at 13 44 32](https://github.com/nus-cs2103-AY2324S2/ip/assets/110611841/b9a17896-084b-43f5-8f05-c235151dd27c)

This could be set in IntelliJ preferences -> Editor -> Coding Style -> Java -> Wrapping and Braces -> 'switch' statement -> Indent 'case' branches.

0-yibai commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,97 @@
+        switch (command) {
+            case "bye":
+                return new ExitTaskCommand();
+                //fall through
```
According to our coding standard, `//fall through` is necessary here because there is no `break`. However, it seems that `return` statement here functions just like a `break`. I'm not sure if `//fall through` comment is necessary here. Maybe try clarify this in forum?

0-yibai commented on src/main/java/duke/command/DeleteTaskCommand.java:
```diff
@@ -0,0 +1,45 @@
+    /**
+     * Executes the command by deleting the specified task from the task list,
+     * saving the updated task list to storage, and displaying a message indicating
+     * the task has been deleted.
```
The method summary seems a bit long to me. Maybe break the parts after the first comma into another sentence?

From our coding standard:
![Screenshot 2024-02-07 at 14 03 49](https://github.com/nus-cs2103-AY2324S2/ip/assets/110611841/25c45da6-c2f6-4111-96f2-71ffe310b506)

0-yibai commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,94 @@
+        return "[E]" + super.toString()
+                + " (from: " + fromDateTime.format(DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm"))
+                + " to: " + toDateTime.format(DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm")) + ")";
+    }
```
I like your use of line breaks to improve readability :thumbsup:

0-yibai commented on src/main/java/duke/task/ToDo.java:
```diff
@@ -0,0 +1,47 @@
+     */
+    @Override
+    public String toFileString() {
+        return "T | " + (getIsDone() ? "1" : "0") + " | " + getDescription();
```
I like your use of white spaces within a statement :thumbsup:

0-yibai reviewed (commented):
LGTM. Overall high quality code adhering to coding style standard with super detailed documentation. Good Job! Just a few minor coding style issues to note 😺

kinjalagarwal1810 committed:
Edited according to pull requests

kinjalagarwal1810 committed:
Implemented A-Checkstyle

kinjalagarwal1810 committed:
Merge branch 'branch-A-CheckStyle'

kinjalagarwal1810 committed:
Edited build.gradle and Duke.java

kinjalagarwal1810 committed:
Implemented Level-10

kinjalagarwal1810 committed:
Merge branch 'branch-Level-10'

kinjalagarwal1810 committed:
Edited Duke.java

kinjalagarwal1810 committed:
Edited the implemention of Level-10

kinjalagarwal1810 committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	src/main/java/duke/Duke.java

kinjalagarwal1810 committed:
Edited Duke.java

kinjalagarwal1810 committed:
Merge branch 'branch-Level-10'

kinjalagarwal1810 committed:
Added GuiObserver.java

kinjalagarwal1810 committed:
Edited files for Level-10

kinjalagarwal1810 committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	src/main/java/duke/GuiObserver.java
#	src/main/java/duke/Main.java

kinjalagarwal1810 committed:
Edited Duke.java

kinjalagarwal1810 committed:
Edited for Level-10

kinjalagarwal1810 committed:
Comppleted Level-10

kinjalagarwal1810 committed:
Merge branch 'branch-Level-10' of https://github.com/kinjalagarwal1810/ip into branch-Level-10

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Main.java

kinjalagarwal1810 committed:
Edit Level-10

kinjalagarwal1810 committed:
Generate a new JAR file

kinjalagarwal1810 committed:
Implement assertions across Duke application for enhanced error detection

- Introduce assertions in `TaskList` for validating task operations and ensuring non-null task list initialization.
- Add input validation assertions in `Ui` methods to verify non-null parameters for task display and interaction messages.
- Ensure `Storage` class methods operate with valid file paths and task data through assertions.
- Strengthen `Duke` class by asserting proper initialization and operation flows.

This update improves the usability of the Duke application by employing assertions to catch and document assumptions, prevent invalid states, and facilitate easier debugging and maintenance.

kinjalagarwal1810 committed:
Merge pull request #1 from kinjalagarwal1810/branch-A-Assertions

Implement assertions across Duke application for enhanced error detec…

kinjalagarwal1810 committed:
Implement A-CodeQuality

kinjalagarwal1810 committed:
Merge pull request #2 from kinjalagarwal1810/branch-A-CodeQuality

Implement A-CodeQuality

kinjalagarwal1810 committed:
Implement C-Update Extension

kinjalagarwal1810 committed:
Edit a few minor bugs and add UserGuide

kinjalagarwal1810 committed:
Edit README

kinjalagarwal1810 committed:
Edit README

kinjalagarwal1810 committed:
Edit build.gradle

kinjalagarwal1810 committed:
Edit build.gradle
